### ISSUE TITLE
Dev: [Time] PHP 8.2.4 SIGSEGV

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#7362]
See https://github.com/codeigniter4/CodeIgniter4/pull/7358#issuecomment-1474679180

The following tests cause `Segmentation fault: 11`:
- tests/system/I18n/TimeLegacyTest.php
- tests/system/I18n/TimeTest.php

<details>

```
$ vendor/bin/phpunit tests/system/I18n/TimeTest.php --debug
PHPUnit 9.6.5 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.4
Configuration: /Users/kenji/work/codeigniter/official/CodeIgniter4/phpunit.xml

Test 'CodeIgniter\I18n\TimeTest::testNewTimeNow' started
Test 'CodeIgniter\I18n\TimeTest::testNewTimeNow' ended
Test 'CodeIgniter\I18n\TimeTest::testTimeWithTimezone' started
Test 'CodeIgniter\I18n\TimeTest::testTimeWithTimezone' ended
Test 'CodeIgniter\I18n\TimeTest::testTimeWithTimezoneAndLocale' started
Test 'CodeIgniter\I18n\TimeTest::testTimeWithTimezoneAndLocale' ended
Test 'CodeIgniter\I18n\TimeTest::testTimeWithDateTimeZone' started
Test 'CodeIgniter\I18n\TimeTest::testTimeWithDateTimeZone' ended
Test 'CodeIgniter\I18n\TimeTest::testToDateTime' started
Test 'CodeIgniter\I18n\TimeTest::testToDateTime' ended
Test 'CodeIgniter\I18n\TimeTest::testNow' started
Test 'CodeIgniter\I18n\TimeTest::testNow' ended
Test 'CodeIgniter\I18n\TimeTest::testParse' started
Test 'CodeIgniter\I18n\TimeTest::testParse' ended
Test 'CodeIgniter\I18n\TimeTest::testToDateTimeString' started
Test 'CodeIgniter\I18n\TimeTest::testToDateTimeString' ended
Test 'CodeIgniter\I18n\TimeTest::testToDateTimeStringWithTimeZone' started
Test 'CodeIgniter\I18n\TimeTest::testToDateTimeStringWithTimeZone' ended
Test 'CodeIgniter\I18n\TimeTest::testToday' started
Test 'CodeIgniter\I18n\TimeTest::testToday' ended
Test 'CodeIgniter\I18n\TimeTest::testTodayLocalized' started
Test 'CodeIgniter\I18n\TimeTest::testTodayLocalized' ended
Test 'CodeIgniter\I18n\TimeTest::testYesterday' started
Test 'CodeIgniter\I18n\TimeTest::testYesterday' ended
Test 'CodeIgniter\I18n\TimeTest::testTomorrow' started
Test 'CodeIgniter\I18n\TimeTest::testTomorrow' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromDate' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromDate' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromDateLocalized' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromDateLocalized' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTime' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTime' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimeEvening' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimeEvening' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimeLocalized' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimeLocalized' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormat' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormat' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormatWithTimezoneString' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormatWithTimezoneString' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormatWithTimezoneObject' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormatWithTimezoneObject' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormatWithInvalidFormat' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromFormatWithInvalidFormat' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimestamp' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimestamp' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimestampWithTimezone' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromTimestampWithTimezone' ended
Test 'CodeIgniter\I18n\TimeTest::testTestNow' started
Test 'CodeIgniter\I18n\TimeTest::testTestNow' ended
Test 'CodeIgniter\I18n\TimeTest::testMagicIssetTrue' started
Test 'CodeIgniter\I18n\TimeTest::testMagicIssetTrue' ended
Test 'CodeIgniter\I18n\TimeTest::testMagicIssetFalse' started
Test 'CodeIgniter\I18n\TimeTest::testMagicIssetFalse' ended
Test 'CodeIgniter\I18n\TimeTest::testGetYear' started
Test 'CodeIgniter\I18n\TimeTest::testGetYear' ended
Test 'CodeIgniter\I18n\TimeTest::testGetMonth' started
Test 'CodeIgniter\I18n\TimeTest::testGetMonth' ended
Test 'CodeIgniter\I18n\TimeTest::testGetDay' started
Test 'CodeIgniter\I18n\TimeTest::testGetDay' ended
Test 'CodeIgniter\I18n\TimeTest::testGetHour' started
Test 'CodeIgniter\I18n\TimeTest::testGetHour' ended
Test 'CodeIgniter\I18n\TimeTest::testGetMinute' started
Test 'CodeIgniter\I18n\TimeTest::testGetMinute' ended
Test 'CodeIgniter\I18n\TimeTest::testGetSecond' started
Test 'CodeIgniter\I18n\TimeTest::testGetSecond' ended
Test 'CodeIgniter\I18n\TimeTest::testGetDayOfWeek' started
Test 'CodeIgniter\I18n\TimeTest::testGetDayOfWeek' ended
Test 'CodeIgniter\I18n\TimeTest::testGetDayOfYear' started
Test 'CodeIgniter\I18n\TimeTest::testGetDayOfYear' ended
Test 'CodeIgniter\I18n\TimeTest::testGetWeekOfMonth' started
Test 'CodeIgniter\I18n\TimeTest::testGetWeekOfMonth' ended
Test 'CodeIgniter\I18n\TimeTest::testGetWeekOfYear' started
Test 'CodeIgniter\I18n\TimeTest::testGetWeekOfYear' ended
Test 'CodeIgniter\I18n\TimeTest::testGetTimestamp' started
Test 'CodeIgniter\I18n\TimeTest::testGetTimestamp' ended
Test 'CodeIgniter\I18n\TimeTest::testGetAge' started
Test 'CodeIgniter\I18n\TimeTest::testGetAge' ended
Test 'CodeIgniter\I18n\TimeTest::testAgeNow' started
Test 'CodeIgniter\I18n\TimeTest::testAgeNow' ended
Test 'CodeIgniter\I18n\TimeTest::testAgeFuture' started
Test 'CodeIgniter\I18n\TimeTest::testAgeFuture' ended
Test 'CodeIgniter\I18n\TimeTest::testGetAgeSameDayOfBirthday' started
Test 'CodeIgniter\I18n\TimeTest::testGetAgeSameDayOfBirthday' ended
Test 'CodeIgniter\I18n\TimeTest::testGetAgeNextDayOfBirthday' started
Test 'CodeIgniter\I18n\TimeTest::testGetAgeNextDayOfBirthday' ended
Test 'CodeIgniter\I18n\TimeTest::testGetAgeBeforeDayOfBirthday' started
Test 'CodeIgniter\I18n\TimeTest::testGetAgeBeforeDayOfBirthday' ended
Test 'CodeIgniter\I18n\TimeTest::testGetQuarter' started
Test 'CodeIgniter\I18n\TimeTest::testGetQuarter' ended
Test 'CodeIgniter\I18n\TimeTest::testGetDST' started
Test 'CodeIgniter\I18n\TimeTest::testGetDST' ended
Test 'CodeIgniter\I18n\TimeTest::testGetDSTUnobserved' started
Test 'CodeIgniter\I18n\TimeTest::testGetDSTUnobserved' ended
Test 'CodeIgniter\I18n\TimeTest::testGetLocal' started
Test 'CodeIgniter\I18n\TimeTest::testGetLocal' ended
Test 'CodeIgniter\I18n\TimeTest::testGetUtc' started
Test 'CodeIgniter\I18n\TimeTest::testGetUtc' ended
Test 'CodeIgniter\I18n\TimeTest::testGetTimezone' started
Test 'CodeIgniter\I18n\TimeTest::testGetTimezone' ended
Test 'CodeIgniter\I18n\TimeTest::testGetTimezonename' started
Test 'CodeIgniter\I18n\TimeTest::testGetTimezonename' ended
Test 'CodeIgniter\I18n\TimeTest::testSetYear' started
Test 'CodeIgniter\I18n\TimeTest::testSetYear' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMonthNumber' started
Test 'CodeIgniter\I18n\TimeTest::testSetMonthNumber' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMonthLongName' started
Test 'CodeIgniter\I18n\TimeTest::testSetMonthLongName' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMonthShortName' started
Test 'CodeIgniter\I18n\TimeTest::testSetMonthShortName' ended
Test 'CodeIgniter\I18n\TimeTest::testSetDay' started
Test 'CodeIgniter\I18n\TimeTest::testSetDay' ended
Test 'CodeIgniter\I18n\TimeTest::testSetDayOverMaxInCurrentMonth' started
Test 'CodeIgniter\I18n\TimeTest::testSetDayOverMaxInCurrentMonth' ended
Test 'CodeIgniter\I18n\TimeTest::testSetDayNotOverMaxInCurrentMonth' started
Test 'CodeIgniter\I18n\TimeTest::testSetDayNotOverMaxInCurrentMonth' ended
Test 'CodeIgniter\I18n\TimeTest::testSetHour' started
Test 'CodeIgniter\I18n\TimeTest::testSetHour' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMinute' started
Test 'CodeIgniter\I18n\TimeTest::testSetMinute' ended
Test 'CodeIgniter\I18n\TimeTest::testSetSecond' started
Test 'CodeIgniter\I18n\TimeTest::testSetSecond' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMonthTooSmall' started
Test 'CodeIgniter\I18n\TimeTest::testSetMonthTooSmall' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMonthTooBig' started
Test 'CodeIgniter\I18n\TimeTest::testSetMonthTooBig' ended
Test 'CodeIgniter\I18n\TimeTest::testSetDayTooSmall' started
Test 'CodeIgniter\I18n\TimeTest::testSetDayTooSmall' ended
Test 'CodeIgniter\I18n\TimeTest::testSetDayTooBig' started
Test 'CodeIgniter\I18n\TimeTest::testSetDayTooBig' ended
Test 'CodeIgniter\I18n\TimeTest::testSetHourTooSmall' started
Test 'CodeIgniter\I18n\TimeTest::testSetHourTooSmall' ended
Test 'CodeIgniter\I18n\TimeTest::testSetHourTooBig' started
Test 'CodeIgniter\I18n\TimeTest::testSetHourTooBig' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMinuteTooSmall' started
Test 'CodeIgniter\I18n\TimeTest::testSetMinuteTooSmall' ended
Test 'CodeIgniter\I18n\TimeTest::testSetMinuteTooBig' started
Test 'CodeIgniter\I18n\TimeTest::testSetMinuteTooBig' ended
Test 'CodeIgniter\I18n\TimeTest::testSetSecondTooSmall' started
Test 'CodeIgniter\I18n\TimeTest::testSetSecondTooSmall' ended
Test 'CodeIgniter\I18n\TimeTest::testSetSecondTooBig' started
Test 'CodeIgniter\I18n\TimeTest::testSetSecondTooBig' ended
Test 'CodeIgniter\I18n\TimeTest::testSetTimezone' started
Test 'CodeIgniter\I18n\TimeTest::testSetTimezone' ended
Test 'CodeIgniter\I18n\TimeTest::testSetTimestamp' started
Test 'CodeIgniter\I18n\TimeTest::testSetTimestamp' ended
Test 'CodeIgniter\I18n\TimeTest::testToDateString' started
Test 'CodeIgniter\I18n\TimeTest::testToDateString' ended
Test 'CodeIgniter\I18n\TimeTest::testToFormattedDateString' started
Test 'CodeIgniter\I18n\TimeTest::testToFormattedDateString' ended
Test 'CodeIgniter\I18n\TimeTest::testToTimeString' started
Test 'CodeIgniter\I18n\TimeTest::testToTimeString' ended
Test 'CodeIgniter\I18n\TimeTest::testCanAddSeconds' started
Test 'CodeIgniter\I18n\TimeTest::testCanAddSeconds' ended
Test 'CodeIgniter\I18n\TimeTest::testCanAddMinutes' started
Test 'CodeIgniter\I18n\TimeTest::testCanAddMinutes' ended
Test 'CodeIgniter\I18n\TimeTest::testCanAddHours' started
Test 'CodeIgniter\I18n\TimeTest::testCanAddHours' ended
Test 'CodeIgniter\I18n\TimeTest::testCanAddDays' started
Test 'CodeIgniter\I18n\TimeTest::testCanAddDays' ended
Test 'CodeIgniter\I18n\TimeTest::testCanAddMonths' started
Test 'CodeIgniter\I18n\TimeTest::testCanAddMonths' ended
Test 'CodeIgniter\I18n\TimeTest::testCanAddMonthsOverYearBoundary' started
Test 'CodeIgniter\I18n\TimeTest::testCanAddMonthsOverYearBoundary' ended
Test 'CodeIgniter\I18n\TimeTest::testCanAddYears' started
Test 'CodeIgniter\I18n\TimeTest::testCanAddYears' ended
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractSeconds' started
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractSeconds' ended
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractMinutes' started
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractMinutes' ended
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractHours' started
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractHours' ended
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractDays' started
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractDays' ended
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractMonths' started
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractMonths' ended
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractYears' started
Test 'CodeIgniter\I18n\TimeTest::testCanSubtractYears' ended
Test 'CodeIgniter\I18n\TimeTest::testEqualWithDifferent' started
Test 'CodeIgniter\I18n\TimeTest::testEqualWithDifferent' ended
Test 'CodeIgniter\I18n\TimeTest::testEqualWithSame' started
Test 'CodeIgniter\I18n\TimeTest::testEqualWithSame' ended
Test 'CodeIgniter\I18n\TimeTest::testEqualWithDateTime' started
Test 'CodeIgniter\I18n\TimeTest::testEqualWithDateTime' ended
Test 'CodeIgniter\I18n\TimeTest::testEqualWithSameDateTime' started
Test 'CodeIgniter\I18n\TimeTest::testEqualWithSameDateTime' ended
Test 'CodeIgniter\I18n\TimeTest::testEqualWithString' started
Test 'CodeIgniter\I18n\TimeTest::testEqualWithString' ended
Test 'CodeIgniter\I18n\TimeTest::testEqualWithStringAndNotimezone' started
Test 'CodeIgniter\I18n\TimeTest::testEqualWithStringAndNotimezone' ended
Test 'CodeIgniter\I18n\TimeTest::testSameSuccess' started
Test 'CodeIgniter\I18n\TimeTest::testSameSuccess' ended
Test 'CodeIgniter\I18n\TimeTest::testSameFailure' started
Test 'CodeIgniter\I18n\TimeTest::testSameFailure' ended
Test 'CodeIgniter\I18n\TimeTest::testSameSuccessAsString' started
Test 'CodeIgniter\I18n\TimeTest::testSameSuccessAsString' ended
Test 'CodeIgniter\I18n\TimeTest::testSameFailAsString' started
Test 'CodeIgniter\I18n\TimeTest::testSameFailAsString' ended
Test 'CodeIgniter\I18n\TimeTest::testBefore' started
Test 'CodeIgniter\I18n\TimeTest::testBefore' ended
Test 'CodeIgniter\I18n\TimeTest::testAfter' started
Test 'CodeIgniter\I18n\TimeTest::testAfter' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeYearsSingle' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeYearsSingle' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeYearsPlural' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeYearsPlural' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeYearsForward' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeYearsForward' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMonthsSingle' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMonthsSingle' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMonthsPlural' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMonthsPlural' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMonthsForward' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMonthsForward' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysSingle' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysSingle' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysPlural' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysPlural' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysForward' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysForward' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysTomorrow' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysTomorrow' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysYesterday' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeDaysYesterday' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeHoursAsTime' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeHoursAsTime' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeHoursAWhileAgo' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeHoursAWhileAgo' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMinutesSingle' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMinutesSingle' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMinutesPlural' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMinutesPlural' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMinutesForward' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeMinutesForward' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWeeksSingle' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWeeksSingle' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWeeksPlural' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWeeksPlural' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWeeksForward' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWeeksForward' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeNow' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeNow' ended
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWithArLocale' started
Test 'CodeIgniter\I18n\TimeTest::testHumanizeWithArLocale' ended
Test 'CodeIgniter\I18n\TimeTest::testSetTimezoneDate' started
Test 'CodeIgniter\I18n\TimeTest::testSetTimezoneDate' ended
Test 'CodeIgniter\I18n\TimeTest::testCreateFromInstance' started
Test 'CodeIgniter\I18n\TimeTest::testCreateFromInstance' ended
Test 'CodeIgniter\I18n\TimeTest::testGetter' started
Test 'CodeIgniter\I18n\TimeTest::testGetter' ended
Test 'CodeIgniter\I18n\TimeTest::testUnserializeTimeObject' started
Test 'CodeIgniter\I18n\TimeTest::testUnserializeTimeObject' ended
Test 'CodeIgniter\I18n\TimeTest::testSetTestNowWithTimeZone' started
Segmentation fault: 11
```
</details>

**Comment by paulbalandan** [comment]
Seems to be from unserialization:

```console
$ vendor/bin/phpunit --filter TimeTest::testUnserializeTimeObject --debug
PHPUnit 9.6.5 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.4 with Xdebug 3.2.0
Configuration: C:\Users\P\Desktop\Web Dev\CodeIgniter4\phpunit.xml.dist

Test 'CodeIgniter\I18n\TimeTest::testUnserializeTimeObject' started
Segmentation fault
```

**Comment by kenjis** [comment]
Yes, it would be better to lock the maximum version.
Otherwise, the GitHub action checks always fail.

**Comment by kenjis** [comment]
```diff
--- a/tests/system/I18n/TimeTest.php
+++ b/tests/system/I18n/TimeTest.php
@@ -1131,11 +1131,11 @@ final class TimeTest extends CIUnitTestCase
     {
         $time1     = new Time('August 28, 2020 10:04:00pm', 'Asia/Manila', 'en');
         $timeCache = serialize($time1);
-        $time2     = unserialize($timeCache);
-
-        $this->assertInstanceOf(Time::class, $time2);
-        $this->assertTrue($time2->equals($time1));
-        $this->assertNotSame($time1, $time2);
+//        $time2     = unserialize($timeCache);
+//
+//        $this->assertInstanceOf(Time::class, $time2);
+//        $this->assertTrue($time2->equals($time1));
+//        $this->assertNotSame($time1, $time2);
     }
 
     public function testSetTestNowWithTimeZone()
```
```console
$ vendor/bin/phpunit tests/system/I18n/TimeTest.php 
PHPUnit 9.6.5 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.4
Configuration: /Users/kenji/work/codeigniter/official/CodeIgniter4/phpunit.xml

..................................................................................................... 101 / 133 ( 75%)
........................Segmentation fault: 11
```
But
```console
bash-3.2$ cat test.php 
<?php

use CodeIgniter\I18n\Time;

require __DIR__ . '/vendor/autoload.php';

$time1     = new Time('August 28, 2020 10:04:00pm', 'Asia/Manila', 'en');
$timeCache = serialize($time1);
bash-3.2$ php test.php 
bash-3.2$ 
```

**Comment by kenjis** [comment]
```console
$ vendor/bin/phpunit tests/system/I18n/TimeTest.php
PHPUnit 9.6.6 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.5
Configuration: /Users/kenji/work/codeigniter/official/CodeIgniter4/phpunit.xml

........................................................................................................................ 120 / 133 ( 90%)
......Segmentation fault: 11
```

**Comment by LeMyst** [comment]
Does the segmentation fault appear in 8.2.7 too?

**Comment by kenjis** [comment]
```
$ vendor/bin/phpunit tests/system/I18n/TimeTest.php
Failed loading /usr/local/Cellar/php/8.2.5/pecl/20220829/xdebug.so:  dlopen(/usr/local/Cellar/php/8.2.5/pecl/20220829/xdebug.so, 0x0009): tried: '/usr/local/Cellar/php/8.2.5/pecl/20220829/xdebug.so' (no such file)
PHPUnit 9.6.9 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.7
Configuration: /Users/kenji/work/codeigniter/official/CodeIgniter4/phpunit.xml

.......................................................................................................... 106 / 133 ( 79%)
....................Segmentation fault: 11
```

**Comment by neznaika0** [comment]
php 8.3 should fix serialization errors, maybe it will help with segfault
https://wiki.php.net/rfc/improve_unserialize_error_handling

**Comment by kenjis** [comment]
The bug was fixed in PHP 8.2.8?

```diff
--- a/tests/system/I18n/TimeTest.php
+++ b/tests/system/I18n/TimeTest.php
@@ -1126,17 +1126,16 @@ final class TimeTest extends CIUnitTestCase
         $this->assertNull($time->weekOfWeek);
     }
 
-    // @TODO Uncomment when PHP 8.2.4 Segmentation fault fixed.
-    //    public function testUnserializeTimeObject()
-    //    {
-    //        $time1     = new Time('August 28, 2020 10:04:00pm', 'Asia/Manila', 'en');
-    //        $timeCache = serialize($time1);
-    //        $time2     = unserialize($timeCache);
-    //
-    //        $this->assertInstanceOf(Time::class, $time2);
-    //        $this->assertTrue($time2->equals($time1));
-    //        $this->assertNotSame($time1, $time2);
-    //    }
+    public function testUnserializeTimeObject()
+    {
+        $time1     = new Time('August 28, 2020 10:04:00pm', 'Asia/Manila', 'en');
+        $timeCache = serialize($time1);
+        $time2     = unserialize($timeCache);
+
+        $this->assertInstanceOf(Time::class, $time2);
+        $this->assertTrue($time2->equals($time1));
+        $this->assertNotSame($time1, $time2);
+    }
 
     public function testSetTestNowWithTimeZone(): void
     {
```

```console
bash-3.2$ vendor/bin/phpunit tests/system/I18n/TimeTest.php 
PHPUnit 9.6.10 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.8
Configuration: /Users/kenji/work/codeigniter/official/CodeIgniter4/phpunit.xml

...........................................................................................  91 / 133 ( 68%)
..........................................                                                  133 / 133 (100%)

Time: 00:00.648, Memory: 20.00 MB

OK (133 tests, 196 assertions)
```

**Comment by neznaika0** [comment]
I don't have version 8.2.8, the error should occur in the entire 8.2.* branch or maybe older.
```Shell
$ vendor/bin/phpunit tests/system/I18n/TimeTest.php 
PHPUnit 9.6.10 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.7
Configuration: /home/aleksandr/www/ci-app/phpunit.xml.dist
Warning:       No code coverage driver available

.............................................................................................................................. 126 / 133 ( 94%)
.......                                                                                                                        133 / 133 (100%)

Time: 00:00.235, Memory: 16.00 MB

OK (133 tests, 196 assertions)
```

**Comment by kenjis** [comment]
Oh, I see that the test passed on PHP 8.2.7. 
I got Segmentation fault on my macOS environment, though.
https://github.com/codeigniter4/CodeIgniter4/issues/7362#issuecomment-1596303135

**Comment by neznaika0** [comment]
Hm, my system
`Linux aleksandr-debian 6.1.0-10-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.38-1 (2023-07-14) x86_64 GNU/Linux`

**Comment by LeMyst** [comment]
I tried on my website with PHP 8.2.8 from sury debian repository.
It seems my issue doesn't appear anymore.

php8.2-common (8.2.8-1+0~20230716.25+debian11~1.gbp5cf2f6)

**Comment by neznaika0** [comment]
Carbon does something similar to what I sent to PR
https://github.com/briannesbitt/Carbon/blob/4308217830e4ca445583a37d1bf4aff4153fa81c/src/Carbon/Traits/Serialization.php#L146

**Comment by kenjis** [comment]
@LeMyst That's it!

**Comment by kenjis** [comment]
@neznaika0 
> In response to this bug, nesbot/carbon has a workaround that implements a userland serialization algorithm, which significantly worse performance. (https://github.com/briannesbitt/Carbon/pull/2644)
https://github.com/php/php-src/issues/11455

Fix serialization for PHP 8.2 
https://github.com/briannesbitt/Carbon/pull/2644/commits/c9cf89f51e15278d6396965e4fcf1c2499774000

**Comment by kenjis** [comment]
This PHP bug has been fixed in PHP 8.2.8.
https://github.com/php/php-src/commit/93becab506ac05a7bddce1ceaacb53d6657180ce
